core/Test/Tasty/Run.hs

```diff
@@ -219,14 +221,21 @@ type Tr = Traversal
 
 -- | Exceptions related to dependencies between tests.
 data DependencyException
-  = DependencyLoop
+  = DependencyLoop [[Path]]
     -- ^ Test dependencies form a loop. In other words, test A cannot start
     -- until test B finishes, and test B cannot start until test
-    -- A finishes.
+    -- A finishes. Field lists detected cycles.
   deriving (Typeable)
 
 instance Show DependencyException where
-  show DependencyLoop = "Test dependencies form a loop."
+  show (DependencyLoop css) = "Test dependencies form a loop:\n\n" ++ showCycles css
```

Comment:
I see! I didn't realize that the message you posted in the PR description had four cycles. Scratch my previous suggestion. :)
I'd still suggest to change the message to something a little more informative. Something like:

```suggestion
  show (DependencyLoop css) = "Test dependencies have cycles:\n\n" ++ showCycles css
```
This would have helped my understanding of the PR description, for instance.

I'd still be in favor of less characters on the message and a more familiar formatting, something like:
```
Test dependencies have cycles:
- a, b, c, a
- x, y, x
- z, z
```

But that's just a personal preference

Reply:
Makes sense, I'll ping you when it's done.